File: ABC/ABC343/D/main.py
n, t = map(int, input().split())
score = [0] * n
mp = {0: n}
res = 1

for _ in range(t):
    a, b = map(int, input().split())
    a -= 1
    mp[score[a]] -= 1
    if mp[score[a]] == 0:
        res -= 1
    score[a] += b
    if not score[a] in mp:
        mp[score[a]] = 1
    else:
        mp[score[a]] += 1
    if mp[score[a]] == 1:
        res += 1
    print(res)

File: ABC/ABC343/D/main.cc
#include <iostream>
#include <map>
#include <vector>

using namespace std;
using ll = long long;

int main() {
  int n, t;
  cin >> n >> t;

  vector<ll> score(n, 0);
  map<ll, int> mp;
  mp[0] = n;

  int res = 1;
  while (t--) {
    int a;
    ll b;
    cin >> a >> b;
    if (--mp[score[--a]] == 0) --res;
    score[a] += b;
    if (++mp[score[a]] == 1) ++res;
    cout << res << endl;
  }

  return 0;
}